target/linux/ramips/dts/mt7628an_tplink_re305-v1.dts

```diff
@@ -0,0 +1,143 @@
+/dts-v1/;
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "tplink,re305-v1", "mediatek,mt7628an-soc";
+	model = "TP-Link RE305 v1";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &ethernet;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio1 5 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio1 6 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "re305-v1:blue:power";
+			gpios = <&gpio1 7 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "re305-v1:blue:wlan2g";
+			gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		wlan5g {
+			label = "re305-v1:blue:wlan5g";
+			gpios = <&gpio1 8 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+		rssi1 {
+			label = "re305-v1:red:rssi";
+			gpios = <&gpio1 11 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi2 {
+			label = "re305-v1:blue:rssi";
+			gpios = <&gpio1 10 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x20000>;
+				read-only;
+			};
+
+			partition@20000 {
+				compatible = "tplink,firmware";
+				label = "firmware";
+				reg = <0x20000 0x5e0000>;
+			};
+
+			config: partition@600000 {
+				label = "config";
+				reg = <0x600000 0x50000>;
+				read-only;
+			};
+
+			radio: partition@7f0000 {
+				label = "radio";
+				reg = <0x7f0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		ralink,group = "i2s", "refclk", "wdt", "p0led_an", "p1led_an", "p2led_an", "p3led_an", "p4led_an";
+		ralink,function = "gpio";
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	mt76@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&radio 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+		mtd-mac-address = <&config 0x10008>;
+		mtd-mac-address-increment = <2>;
+	};
+};
+
+&wmac {
+	status = "okay";
+	mtd-mac-address = <&config 0x10008>;
+	mediatek,mtd-eeprom = <&radio 0>;
+	mtd-mac-address-increment = <1>;
+};
+
+&ethernet {
+	mtd-mac-address = <&config 0x10008>;
```

Comment:
Typically, addresses in config are written by the vendor firmware and thus shouldn't be used.
However, is it possible that config is only 0x10000 big and you have a "rom" after it? I'm asking because rom 0x8 would be a common tp-link MAC address location.

Reply:
Hello @adrianschmutzler 
in the GPL tarball I can only find these definitions:

total=13,  flash=8M
0=            fs-uboot,0x00000000,0x00020000, 0, 0, 0, 0, u-boot.bin
1=            os-image,0x00020000,0x000f0000, 0, 0, 0, 0, tp.vmlinux.bin.gz
2=         file-system,0x00110000,0x004f0000, 0, 0, 0, 0, rt288x-squashfs
3=     partition-table,0x00600000,0x00002000, 1, 3, 1, 0, #
4=         default-mac,0x00610000,0x00000020, 0, 2, 2, 3, #
5=                 pin,0x00610100,0x00000020, 0, 2, 2, 3, #
6=        product-info,0x00611100,0x00001000, 0, 2, 2, 1, FlashMaker/product-info.bin
7=        soft-version,0x00620000,0x00001000, 3, 2, 2, 0, #
8=        support-list,0x00621000,0x00001000, 0, 2, 2, 0, FlashMaker/supportlist.bin
9=             profile,0x00622000,0x00008000, 0, 2, 2, 0, FlashMaker/profile.bin
10=        user-config,0x00630000,0x00010000, 0, 2, 2, 3, #
11=     default-config,0x00640000,0x00010000, 0, 2, 2, 0, FlashMaker/default-config.bin
12=              radio,0x007f0000,0x00010000, 0, 0, 0, 3, #

And I decided to combine the whole config part to one partitition between 0x600000 and 0x650000.

Reply:
Okay, fine, I've just seen that the RE350-v1 (mt7621) uses the same partitioning and also uses config 0x10008.

Reply:
Did you check that the addresses are assigned in the same manner as on stock firmware?
ethernet label
2.4 GHz label+1
5 GHz label + 2
?

Reply:
Yes the adresses match to the behavoir with stock firmware.